### PR TITLE
Don't auto-login on pulling public routes; pull base from share

### PR DESCRIPTION
**Pull request by ChrisPenner**
## Overview

We don't want to force folks to be logged in on public pulls. We also need to this to allow the initial "Welcome" pull of base.

## Implementation notes

Currently I'm just checking if the first path segment is the literal string "public".
Keep in mind that the server still does any auth checks it needs to.

### CODE REVIEW

File: unison-cli/src/Unison/Auth/HTTPClient.hs [hunk 1]
```diff
@@ -17,11 +15,11 @@ newtype AuthenticatedHttpClient = AuthenticatedHttpClient HTTP.Manager
 
 -- | Returns a new http manager which applies the appropriate Authorization header to
 -- any hosts our UCM is authenticated with.
-newAuthenticatedHTTPClient :: MonadIO m => (Output v -> IO ()) -> TokenProvider -> UCMVersion -> m AuthenticatedHttpClient
-newAuthenticatedHTTPClient responder tokenProvider ucmVersion = liftIO $ do
+newAuthenticatedHTTPClient :: MonadIO m => TokenProvider -> UCMVersion -> m AuthenticatedHttpClient
```
Comment (mitchellwrosen) [inline unison-cli/src/Unison/Auth/HTTPClient.hs:18]: Why don't we want to print these `CredentialFailureMsg`s anymore?

File: unison-cli/src/Unison/Codebase/Editor/HandleInput.hs [hunk 2]
```diff
@@ -2315,10 +2315,11 @@ viewRemoteGitBranch ns gitBranchBehavior action = do
   eval $ ViewRemoteGitBranch ns gitBranchBehavior action
 
 importRemoteShareBranch :: MonadUnliftIO m => ReadShareRemoteNamespace -> Action' m v (Either (Output v) (Branch m))
-importRemoteShareBranch ReadShareRemoteNamespace {server, repo, path} = do
+importRemoteShareBranch (rrn@(ReadShareRemoteNamespace {server, repo, path})) = do
```
Comment (mitchellwrosen) [inline unison-cli/src/Unison/Codebase/Editor/HandleInput.hs]:
```suggestion
importRemoteShareBranch rrn@(ReadShareRemoteNamespace {server, repo, path}) = do
```

File: unison-cli/tests/Unison/Test/VersionParser.hs [hunk 3]
```diff
@@ -27,9 +27,10 @@ makeTest (version, path) =
       (rightMay $ runParser defaultBaseLib "versionparser" version)
       ( Just
           -- We've hard-coded the v4 branch for base for now. See 'defaultBaseLib'
```
Comment (mitchellwrosen) [inline unison-cli/tests/Unison/Test/VersionParser.hs]: This comment is out of date now. Also, are these changes to where we download `base` being pulled in from another PR, or is it supposed to be a part of this one?

**Review by mitchellwrosen — APPROVED** [review]

File: unison-cli/src/Unison/Auth/HTTPClient.hs (hunk 1 above)
Comment (ChrisPenner) [reply unison-cli/src/Unison/Auth/HTTPClient.hs:18]: Since this is location is triggered pre-request, we print a ton of "you need to re-auth" messages when pulling from public endpoints when really the user doesn't need to be authed for those at all.

There's a check in the response which triggers a "You need to re-auth" if the result is a 401 which is more accurate and covers the same situation 👍🏼

File: unison-cli/tests/Unison/Test/VersionParser.hs (hunk 3 above)
Comment (ChrisPenner) [reply unison-cli/tests/Unison/Test/VersionParser.hs]: Yeah sorry I lumped the two together since the whole reason I ended up adding public pulling was to be able to pull base on start-up 😓

Comment (mitchellwrosen) [reply unison-cli/tests/Unison/Test/VersionParser.hs]: Lumping is cool with me
